src/components/navbar/Wallet.tsx

```diff
@@ -79,7 +79,13 @@ const AccountContainer = ({ account }: { account?: string }) => {
   return (
     <div
       className={accountLinkContainer}
-      onClick={async () => await WalletStore.connect(true)}
+      onClick={async () => {
+        if (needNetworkChange) {
```

Comment:
This is boilerplate code, we should but it into `WalletStore`

Reply:
This will look like:
```TS
WalletStore.connect({clearCachedProvider: true, needNetwrokChange })
```

Reply:
I agree that it's a boilerplate and we need to extract it to WalletStore, but I don't think that it's a good idea to put it in `connect` function. The only one thing we need to do is to switch the network, not to connect anything (provider or whatever). I'll create a function like `connectOrChangeNetwork`.

Reply:
Okay, thanks :)